Upload cards as attachment and cleanup cards.js

hgarus opened the pull request:
Fixes #4 .

Not quite sure where to put `longestCommonPrefix`. Definitely doesn't belong on the class, feels like it should be in it's own module, but I'm not sure if we want relative paths all over the place (such as `require("../../helpers/stringhelper")`. An alternative would be using common-prefix from npm.

I have replaced the use of a map to find cards with card images with a slightly different approach using lodash, personally I find it more readable, if you think differently feel free to let me know.

I have tested this by running a selfbot and there is a noticable delay on answer times due to the image upload. I'm not sure if the added cleanliness is actually worth the delay, compared to just adding the cardImage URL to the message.

hgarus committed:
Upload cards as attachment and cleanup cards.js

Restructure card aggregator:
* Split find into two functions findCard and cardToString
* Add lodash dependency and use it in several spots, especially search
* add Manacost to the text output
* extend the existing respondToMsg callback to allow file uploads

hcschild commented on aggregators/mtg_cards/cards.js:
```diff
@@ -1,87 +1,79 @@
+        let card = uniqCards.find(c => c.name.toLowerCase() === cardName);
+        if (!card){
+            // Use a hit which shares the longest common prefix with the search term
+            card = _.maxBy(uniqCards, c => longestCommonPrefix(cardName, c.name.toLowerCase()));
```
Is this better then just `uniqCards.find(c => c.name.toLowerCase().startsWith(cardName));`?

hgarus replied:
Maybe I've been overthinking this. You're right that there probably aren't any search results where we only have a partial match on the prefix and still want to prefer that result.

hcschild commented on aggregators/mtg_cards/cards.js:
```diff
@@ -1,87 +1,79 @@
+                if (otherCardNames.length) {
+                    otherCardNames.sort();
+                    otherCardNames = _.sortedUniq(otherCardNames);
+                    response += "\n\nOther matching cards: " + otherCardNames.join(", ");
```
It would be better to `join` the card names with `join(" | ")` because some card names contain commas.

bra1n replied:
I'd also put `*` around the card names.

hcschild commented on aggregators/mtg_cards/cards.js:
```diff
@@ -1,87 +1,79 @@
 
-    getContent(command , parameter, callback) {
-        this.find(parameter.toLowerCase(), callback);
+function longestCommonPrefix(...strings) {
```
As I said in my comment for line `39` I don't know if this is needed.

bra1n replied:
Probably not needed, or if it really was, then not as an external function. ;)

hcschild reviewed (changes requested):
This looks goods except for some minor issues. Additionally you need to update your package.json

hgarus committed:
Make some requested changes

Get rid of the common prefix stuff, it's unnecessary.

Use | to make the list of other matching cards better readable.

bra1n commented:
Looks good to me now.
